### DISCUSSION TITLE
how much space needed for a default runner

### DISCUSSION THREAD

**Discussion by roshvin** [actions/actions-runner-controller#1456] [Questions] [answered]
when i was running 100 runner at a time i got space issue . my kubernetes have only 750GB storage attached.

Run actions/setup-java@v1
Error: ENOSPC: no space left on device, write

by default how much space is needed for one runner , based on it i will put max runner value in HRA

**Comment by mumoshu** [comment, marked as answer]
Please check my answer at https://github.com/actions-runner-controller/actions-runner-controller/issues/1459#issuecomment-1131430057.

Which storage medium are you using for emptyDir in your K8s cluster? Is 750GB the size of the root volume, or the secondary volume?
AFAIK, emptyDirs are mapped to directories under `/var/lib/kubelet/pods/` on your host. You'll probably need to ensure that your large volume is mounted at `/var` or `/var/lib` or `/var/lib/kubelet` or etc.

**Reply by roshvin** [reply]
Hi @mumoshu   we have 750 GB of storage class attached to our Kubernetes cluster . 
normally for other applications we create PV and PVC and mount the PVC to the volume inside container. 

with ARC could you suggest how we can  mount the PV or PVC inside action runner

**Reply by mumoshu** [reply]
You should try RunnerSet which supports VolumeClaimTemplates similar to StatefulSet.

**Reply by roshvin** [reply]
Hi @mumoshu  i red through documentation and couldn't find any template of RunnerSet with pvc & PV . could you pls give an example runnerset with a pvc .

**Reply by mumoshu** [reply]
@roshvin I think you have a lot of difficulties figuring out how K8s and ARC work and configuring it appropriately yourself. To be fair, probably it's good timing for you to [consider sponsoring](https://github.com/sponsors/actions-runner-controller) our work so that we can put more effort into improving documentation.

You might also consider [contracting with one of our maintainers](https://github.com/actions-runner-controller/actions-runner-controller/blob/master/CODEOWNERS) or other [contributors](https://github.com/actions-runner-controller/actions-runner-controller/graphs/contributors) so that you can receive more direct support.

We're developing ARC open-source but that doesn't mean we can afford free user support to every enterprise user with very complex requirements. Thanks for your understanding :pray:

Anyway, I've recently started writing https://github.com/actions-runner-controller/actions-runner-controller/pull/1464 which might help your use-case too.

**Reply by roshvin** [reply]
Hi @mumoshu  . thanks for the documentation.  i could setup runner setup following the doc in the given issue.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
